definitions/marts/looker_studio_marts/ecf/ls_ecf_correct_cohort_checker.sqlx

```diff
@@ -0,0 +1,255 @@
+        partnership_relationship: "TRUE/FALSE for the partnership_id taken from partnerships_latest_cpd. The relationship is 'live' when the value returns FALSE.",
+        earliest_declaration_date: "The first declaration date for the participant_profile_id where the declaration is funded (eligible, payable or paid).",
+        unique_participant_counter: "1 or 0 if the row is the first instance of the participant_profile_id. Used in reporting as a means of gaining unique count of participants.",
+        has_induction_start_date: "TRUE/FALSE if the induction_start_date field is not null."
+    }
+}
+
+/*** ECF Cohort Corrections Looker Studio Data Mart ***/
+
+/*
+  Objective:
+    To gather all ECTs & Mentors inducted within 2021-2023 cohorts for evaluation as to whether they are allocated to correct the cohort based on criteria below.
+
+  Instructions:
+    For ECTs in the "wrong" cohort with declarations:
+      WHERE
+        eligible_for_funding = TRUE
+        AND
+        participant_role = 'ECT'
+        AND 
+        declaration_id IS NOT NULL
+        AND
+        niot_partnership_chk = 0
+        AND
+        start_date_suggestion_match = 0
+      
+    For Mentors in the "wrong" cohort with declarations:
+      WHERE
+        eligible_for_funding = TRUE
+        AND
+        participant_role = 'Mentor'
+        AND 
+        declaration_id IS NOT NULL
+        AND
+        niot_partnership_chk = 0
+        AND
+        earliest_dec_suggestion_match = 0
+
+    For ECTs in the "wrong" cohort without declarations:
+      WHERE
+        eligible_for_funding = TRUE
+        AND
+        participant_role = 'ECT'
+        AND 
+        declaration_id IS NULL
+        AND
+        niot_partnership_chk = 0
+        AND
+        start_date_suggestion_match = 0
+
+  Dependencies:
+    dataform.ecf_golden_thread
+    dataform.ls_declarations_provider_names
+    dataform.partnerships_latest_cpd
+
+  Version History:
+    1.0 2024-06-05 Tony Page
+      Initial Build
+*/
+
+-- Gather ECF Golden Thread
+WITH golden_thread AS( 
+  SELECT
+     -- Manually assign the user_id for specific participant profile where the profile no longer exists in latest_cpd data and has no superceeded record to call upon.
+     CASE 
+      WHEN egt.participant_profile_id = '8cd0928c-ecad-482f-95cd-6a0ff3e840ad' THEN '79dee4de-a6dd-407c-ab1c-690fcb6d0056'
+      ELSE egt.user_id
+     END AS user_id
+    ,egt.participant_profile_id
+    ,CASE
+      WHEN egt.participant_type LIKE '%ECT' THEN 'ECT'
+      WHEN egt.participant_type LIKE '%Mentor' THEN 'Mentor'
+      ELSE NULL
+     END AS participant_role
+    ,egt.schedule_identifier
+    ,egt.induction_start_date AS induction_start_date
+    ,egt.cohort
+    ,egt.eligible_for_funding
+    ,egt.partnership_id
+    ,prt.relationship AS partnership_relationship
+    ,egt.lead_provider_name AS induction_lead_provider_name
+  FROM 
+    ${ref('ecf_golden_thread')} AS egt
+  LEFT JOIN

```

Comment:
Instead of joining this table here, I'd update the ecf_inductions mart and bring in the relationship field there so that it comes through automatically here.